src/vtpty/SshSession.h

```diff
@@ -0,0 +1,199 @@
+        Connect,                            // connect to SSH server (usually TCP/IPv4 or TCP/IPv6)
+        Handshake,                          // SSH handshake
+        VerifyHostKey,                      // verify host key against known_hosts file
+        AuthenticateAgent,                  // authenticate with SSH agent
+        AuthenticatePrivateKeyStart,        // start private key authentication
+        AuthenticatePrivateKeyWaitForInput, // wait for private key's password (user input))
+        AuthenticatePrivateKey,             // authenticate with private key
+        AuthenticatePasswordStart,          // start password authentication
+        AuthenticatePasswordWaitForInput,   // wait for password (user input)
+        AuthenticatePassword,               // authenticate with password
+        OpenChannel,                        // open SSH channel
+        RequestAuthAgent,                   // request SSH (auth) agent forwarding
+        RequestPty,                         // request pty
+        SetEnv,                             // set environment variables
+        StartShell,                         // shell is starting
+        Operational,                        // setup done, and shell is running
+        ResizeScreen,                       // resize screen, then move to operational
+        Failure,                            // connection closed with protocol related error
+        Closed,                             // connection closed by peer or us
+    };
+
+    // Wait for the socket to become readable and/or writable.
+    int waitForSocket(std::optional<std::chrono::milliseconds> timeout = std::nullopt);
+
+  private:
+    bool connect(std::string_view host, int port);
+    void logErrorWithDetails(int libssl2ErrorCode, std::string_view message) const;
+    bool verifyHostKey();
+    bool authenticateWithAgent();
+    void authenticateWithPrivateKey();
+    bool authenticateWithPassword();
+
+    void handlePreAuthenticationPasswordInput(std::string_view buf, State next);
+
+    void injectRead(std::string_view buf);
+    void injectWrite(std::string_view buf);
+
+    // Handles each individual states.
+    void processState();
+
+    void setState(State nextState);
+
+    void logInfo(std::string_view message) const;
+
+    template <typename... Args>
+    void logInfo(fmt::format_string<Args...> fmt, Args&&... args) const
+    {
+        logInfo(fmt::format(fmt, std::forward<Args>(args)...));
+    }
+
+    void logError(std::string_view message) const;
+
+    template <typename... Args>
+    void logError(fmt::format_string<Args...> fmt, Args&&... args) const
+    {
+        logError(fmt::format(fmt, std::forward<Args>(args)...));
+    }
+
+    std::string _host = "localhost";
+    int _port = 22;
+    std::string _username;
+    std::filesystem::path _privateKeyFile;
+    std::filesystem::path _publicKeyFile;
+    std::filesystem::path _knownHostsFile;
```

Comment:
This repeat config.h ssh config definition, can we declare it here and then reuse in config ?